### PR TITLE
Task 1 create models migrations

### PR DESCRIPTION
**Pull request by sku-wtag**
I have created 6 models, and resolved all of mentioned issue and add some test cases of invalid input field. 

1st
2nd
3rd

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Course < ApplicationRecord
+  has_many :enrollments
+  has_many :users, through: :enrollments
+  has_many :lessons
+  validates :title, :description, presence: true
+  validates :title, length: { minimum: 2, maximum: 30 }
+  validates :description, length: { minimum: 5 }
```
Comment (iud-wtag) [inline app/models/course.rb]:
```suggestion
  has_many :enrollments
  has_many :users, through: :enrollments
  has_many :lessons
  
  validates :title, :description, presence: true
  validates :title, length: { minimum: 2, maximum: 30 }
  validates :description, length: { minimum: 5 }
```

File: app/models/lesson.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Lesson < ApplicationRecord
+  belongs_to :course
+  has_many :user_course_progresses
+  validates :title, :description, :score, presence: true
+  validates :title, length: { minimum: 2, maximum: 30 }
+  validates :description, length: { minimum: 5 }
```
Comment (iud-wtag) [inline app/models/lesson.rb]: ditto

File: config/database.yml [hunk 3]
```diff
@@ -59,7 +59,10 @@ development:
 # Do not set this db to the same as development or production.
 test:
   <<: *default
-  database: lms_test
+  database: <%= ENV["DATABASE_NAME"] %>
+  username: <%= ENV["DATABASE_USERNAME"] %>
+  password: <%= ENV["DATABASE_PASSWORD"] %>
+
```
Comment (iud-wtag) [inline config/database.yml]: remove this extra empty line

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { FactoryBot.create(:user) }
+  let(:invalid_user) { FactoryBot.create(:user) }
+  describe 'Model : ' do
+    describe 'check to be a valid ' do
+      it 'factory ' do
```
Comment (iud-wtag) [inline spec/models/user_spec.rb]: any reason to leave extra space in the end of string?

**Review by iud-wtag — APPROVED** [review]
Good Work 🎉 .. LGTM.

File: spec/models/user_spec.rb (hunk 4 above)
Comment (sku-wtag) [reply spec/models/user_spec.rb]: Bhaiya, I didn't notice when I wrote this. Thank you, Bhaiya, for suggesting me. I will fix this.

**Pull request closed without merging**